### PR TITLE
Fix PlugClean error when the default branch has changed

### LINKED ISSUE #1253: PlugClean fails to recover when remote renamed default branch
When [vim-gitgutter](https://github.com/airblade/vim-gitgutter) changed the remote, default branch from `master` to `main` a while ago, my calls to `:PlugClean` started to fail with 

```
Error detected while processing command line..function <SNR>2_clean[13]..<SNR>2_git_validate:
line   36:
E688: More targets than List items
```

The error occurs [here](https://github.com/junegunn/vim-plug/blob/ca0ae0a8b1bd6380caba2d8be43a2a19baf7dbe2/plug.vim#L2349).

Running `git status` in the vim-gitgutter's worktree shows
```
On branch master
Your branch is based on 'origin/master', but the upstream is gone.
  (use "git branch --unset-upstream" to fixup)

nothing to commit, working tree clean
```

The result from the `system` call on the vim-plug quoted line above is:
```
['''git <command> [<revision>...] -- [<file>...]''']
```

The underlying issue triggering this behavior is the outdated symbolic ref `refs/remotes/origin/HEAD`. This does not get updated on a git fetch. A `git remote set-head origin -a` can do the trick. But this does involve another network call to query the remote.

I run `PlugClean!` daily as part of [this setup script](https://github.com/kaihowl/dotfiles/blob/a01f529f8b6a938cf33369df457bb0faec999e65/nvim/install.sh#L40).

------------------------------

https://github.com/kaihowl/dotfiles/blob/a01f529f8b6a938cf33369df457bb0faec999e65/nvim/config--nvim--init.vim.symlink#L100-L167

```
NVIM v0.9.2
Build type: Release
LuaJIT 2.1.1692716794

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/share/nvim"

Run :checkhealth for more info
```

- Type:
    - [?] Bug
    - [?] Enhancement
- OS:
    - [ ] All/Other
    - [ ] Linux
    - [x] macOS
    - [ ] Windows
- Vim:
    - [ ] Terminal Vim
    - [ ] GVim
    - [x] Neovim

### PR DESCRIPTION
**Pull request by junegunn**
Fix #1253

### CODE REVIEW

**Comment by junegunn** [comment]
I was able to trigger the code by removing `.git/refs/heads/master`.

```
x vim-xxx:
    Failed to compare with the origin. The default branch might have changed.
    PlugClean required.
```

**Comment by junegunn** [comment]
See #1005 for the discussion on automatic update of the default branch.

**Comment by adamency** [comment]
@junegunn Hi. I have read the threads, #1005 and #1272, but this fails to answer my question.

Can you please clarify this error message ?

```
"Failed to compare with the origin. The default branch might have changed.\nPlugClean required."
```

As per your own documentation:

```
`PlugClean[!]`                       | Remove unlisted plugins (bang version will clean without prompt)
```

but your error message tells about using `PlugClean` in the context of **installed** plugins, so either your code has changed and `PlugClean` is doing something else/more now, or your error message is wrong or missing some explanation.

Basically, my comment boils down to the fact that I was running `PlugStatus` to check the status of my plugins after the update of vim-plug itself via `PlugUpgrade` (which was supposed to solve the default branch issue), when I got this error message for all my installed plugins. However I do **not** want to remove the unlisted plugins from the filesystem and I still want to correct the `PlugStatus`.

Please provide information to better understand the issue and solve it.

**Comment by junegunn** [comment]
@adamency 

> when I got this error message for all my installed plugins.

All plugins? That doesn't sound right. Please open a new issue explaining the problem, with the plug section of your Vim configuration, screenshots, etc.
